### PR TITLE
allow dropping & replacing main thumbnail

### LINKED ISSUE #206: Choose preview image if desired
It's not clear to me why anyone would want this option, but I had a user request it 🤷‍♂ 

It's technically feasible. For ease of UI we could have the app receive drag-drop events, and if the event happens over a particular thumbnail, we'll just swap in* the image.

\* `ffmpeg` already _can_ handle an image resize, so it shouldn't be too problematic 😅

### PR DESCRIPTION
**Pull request by whyboris**
⚠️ TODO: document the feature in FAQ !!!

Closes #206

### CODE REVIEW

File: src/app/components/home.component.ts [hunk 1]
```diff
@@ -649,24 +649,43 @@ export class HomeComponent implements OnInit, AfterViewInit {
...
 
+  /**
+   * Handle dropping something over an item in the gallery
+   * Used to handle dropping a .jpg file to replace preview!
+   * @param event         drop event - containing path to possible jpg file
+   * @param galleryItem   item in the gallery over which jpg was dropped
+   */
+  droppedSomethingOverVideo(event, galleryItem: ImageElement) {
+    const pathToNewJpg: string = event.dataTransfer.files[0].path;
+    if (pathToNewJpg.endsWith('.jpg') && galleryItem.cleanName !== '*FOLDER*') {
```
Comment (cal2195) [inline src/app/components/home.component.ts]: `.JPG` `.JPEG` `.jPeG`? 😄 
Can we support other image types?

File: main-extract.ts [hunk 2]
```diff
@@ -510,3 +508,61 @@ export function extractFromTheseFiles(
...
+) {
+
+  return new Promise((resolve, reject) => {
+
+    console.log('RESIZING !?!?!?!?');
+
+    const width: number = Math.floor(height * (16 / 9));
+
+    // desired command:
+    // ffmpeg -y -i IN.png -vf "scale=1280:720:force_original_aspect_ratio=decrease,pad=1280:720:(ow-iw)/2:(oh-ih)/2" OUT.jpg
```
Comment (cal2195) [inline main-extract.ts]: I see `.png` here!

File: src/app/components/home.component.ts [hunk 3]
```diff
@@ -675,7 +675,14 @@ export class HomeComponent implements OnInit, AfterViewInit {
    */
   droppedSomethingOverVideo(event, galleryItem: ImageElement) {
     const pathToNewJpg: string = event.dataTransfer.files[0].path;
-    if (pathToNewJpg.endsWith('.jpg') && galleryItem.cleanName !== '*FOLDER*') {
+    if (
+      (
+          pathToNewJpg.endsWith('.jpg')
+       || pathToNewJpg.endsWith('.jpeg')
+       || pathToNewJpg.endsWith('.png')
```
Comment (cal2195) [inline src/app/components/home.component.ts]: Ah, disregard last comment! 👍

File: src/app/components/home.component.ts (hunk 1 above)
Comment (whyboris) [reply src/app/components/home.component.ts]: thanks! I forgot the `toLowercase()` 😅

Comment (whyboris) [reply src/app/components/home.component.ts]: fixed with #256

**Pull request merged**